### PR TITLE
runtime: enable kernel-hashes for SNP confidential container

### PR DESCRIPTION
**Pull request by niteeshkd**
This is required to provide the hashes of kernel, initrd and cmdline needed during the attestation of the coco.

Fixes: #9150

### CODE REVIEW

**Comment by niteeshkd** [comment]
@ryansavino @fitzthum @fidencio

File: versions.yaml [hunk 1]
```diff
@@ -305,7 +305,7 @@ externals:
       package_output_dir: "OvmfX64"
     sev:
       description: "AmdSev build needed for SEV measured direct boot."
-      version: "edk2-stable202302"
+      version: "edk2-stable202402"
```
Comment (fidencio) [inline versions.yaml:308]: Just one comment here.  It seems we can use stable edk2 pretty much evrywhere, as that has support for SEV / SNP / TDX.

Would it be possible to actually keep the versions of those aligned?

**Review by ryansavino — CHANGES_REQUESTED** [review]

File: versions.yaml [hunk 2]
```diff
@@ -305,7 +305,7 @@ externals:
       package_output_dir: "OvmfX64"
     sev:
       description: "AmdSev build needed for SEV measured direct boot."
-      version: "edk2-stable202302"
```
[the remark this answers is not in the document]
Comment (niteeshkd) [reply versions.yaml:308]: @ryansavino , I have created different commit for upgrading ovmf.

**Review by ryansavino — APPROVED** [review]

**Review by fitzthum — APPROVED** [review]

**Comment by fitzthum** [comment]
Hm can we get another pair of eyes here? Maybe @stevenhorsman  or @danmihai1?

**Comment by stevenhorsman** [comment]
I tried re-running the SNP tests on this and they seem to be failing (compared to the last nightly that was successful), so I'd like to see that pass for this PR before approving. Thanks

**Comment by danmihai1** [comment]
> Hm can we get another pair of eyes here? Maybe @stevenhorsman or @danmihai1?

@fitzthum , thanks for the heads-up! However, I am not knowledgeable enough about kernel hashes to be able to help much here. In AKS, we load linux in the Guest VM using IGVM files and related ways of verifying the integrity of the boot chain.

**Comment by ryansavino** [comment]
All checks and required tests passing.

**Comment by fitzthum** [comment]
SNP is failing again

**Comment by ryansavino** [comment]
This PR is ready to go. Required tests passing. Tapping @fidencio @stevenhorsman @wainersm @danmihai1 for review. Thanks!

**Review by wainersm — APPROVED** [review]
Thanks!

File: versions.yaml [hunk 3]
```diff
@@ -324,12 +324,12 @@ externals:
     url: "https://github.com/tianocore/edk2"
     x86_64:
       description: "Vanilla firmware build"
-      version: "edk2-stable202202"
+      version: "edk2-stable202402"
```
[the remark this answers is not in the document]
Comment (fidencio) [reply versions.yaml:327]: We're not relying on our Kata Containers packaged OVMF for TDX, but rather on what the distros are providing.  IOW, TDX is not affected by this change.

**Pull request merged**